modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -552,11 +657,51 @@ using GConstGIEModel = ade::ConstTypedGraph
     , IECallable
     >;
 
-inline IE::Blob::Ptr extractRemoteBlob(IECallContext& ctx, std::size_t i) {
+cv::MediaFrame preprocess_frame_impl(cv::MediaFrame &&in_frame, const std::string &layer_name,
+                                    IECallContext& ctx,
+                                    const cv::util::optional<cv::Rect> &opt_roi,
+                                    cv::MediaFrame* out_keep_alive_frame,
+                                    bool* out_is_preprocessed) {
+    cv::util::optional<cv::gapi::wip::pp_params> param =
+                        ctx.uu.preproc_engine_impl->is_applicable(in_frame);
+    if (param.has_value()) {
+        GAPI_LOG_DEBUG(nullptr, "VPP preprocessing for decoded remote frame will be used");
+        const cv::GFrameDesc& expected_net_input_descr =
+                    ctx.uu.net_input_params.get_param(layer_name);
+        cv::gapi::wip::pp_session pp_sess =
+                    ctx.uu.preproc_engine_impl->initialize_preproc(param.value(),
+                                                                   expected_net_input_descr);
+
+        in_frame = ctx.uu.preproc_engine_impl->run_sync(pp_sess, in_frame, opt_roi);
+
+        if (out_keep_alive_frame != nullptr) {
+            GAPI_LOG_DEBUG(nullptr, "remember preprocessed remote frame to keep it busy from reuse, slot: " <<
+                                    out_keep_alive_frame);
+            *out_keep_alive_frame = in_frame;
+        }
+        if (out_is_preprocessed) {
+            *out_is_preprocessed = true;
+        }
+    } // otherwise it is not suitable frame, then check on other preproc backend or rely on IE plugin
+    return std::move(in_frame);
+}
+
+inline IE::Blob::Ptr extractRemoteBlob(IECallContext& ctx, std::size_t i,
+                                       const std::string &layer_name,
+                                       const cv::util::optional<cv::Rect> &opt_roi,
+                                       cv::MediaFrame* out_keep_alive_frame,
+                                       bool* out_is_preprocessed) {
     GAPI_Assert(ctx.inShape(i) == cv::GShape::GFRAME &&
                 "Remote blob is supported for MediaFrame only");
+    cv::MediaFrame frame = ctx.inFrame(i);
+    if (ctx.uu.preproc_engine_impl) {
+        GAPI_LOG_DEBUG(nullptr, "Try to use preprocessing for decoded remote frame in remote ctx");
+        frame = preprocess_frame_impl(std::move(frame), layer_name, ctx, opt_roi,
```

Comment:
Actually the copy of `frame` should be almost free

Reply:
Do frame really moves into somewhere?
I doubt this backend code *owns* the frame to to move it. It comes as an input, and it may go somewhere else down to the pipeline. It may still work here, though, but may give some false understanding for those reading this code in the future.

Reply:
it moves back :)
Look:
```
cv::MediaFrame frame = ctx.inFrame(i);
frame = preprocess_frame_impl(std::move(frame));
```